source/iface/device.rst

```diff
@@ -195,97 +297,987 @@ Device Info
+
++-----------------------------+
+| Return type: ``uint32_t``   |
++-----------------------------+
+
+.. rubric:: ``sycl::info::device::partition_properties``
+
+Returns the partition properties supported by this SYCL device; a vector
+of ``sycl::info::partition_property``.
+
+An element is returned in this vector only if the device can be partitioned
+into at least two sub devices along that partition property.
+
++----------------------------------------------------------------+
+| Return type: ``std::vector<sycl::info::partition_property>``   |
++----------------------------------------------------------------+
+
+.. rubric:: ``sycl::info::device::partition_affinity_domains``
+
+Returns a std::vector of the partition affinity domains supported by this
+SYCL device when partitioning with
+``sycl::info::partition_property::partition_by_affinity_domain``.
+
+An element is returned in this vector only if the device can be partitioned
+into at least two sub devices along that affinity domain.
+
++-----------------------------------------------------------------------+
+| Return type: ``std::vector<sycl::info::partition_affinity_domain>``   |
++-----------------------------------------------------------------------+
+
+.. rubric:: ``sycl::info::device::partition_type_property``
+
+Returns the partition property of this SYCL device.
+
+If this SYCL device is not a sub device then the return
+value must be ``sycl::info::partition_property::no_partition``,
+otherwise it must be one of the following values:
+``sycl::info::partition_property::partition_equally``,
+``sycl::info::partition_property::partition_by_counts``,
+``sycl::info::partition_property::partition_by_affinity_domain``.
+
++-------------------------------------------------------+
+| Return type: `sycl::info::partition_property`_        |
++-------------------------------------------------------+
+
+.. rubric:: ``sycl::info::device::partition_type_affinity_domain``
+
+Returns the partition affinity domain of this SYCL device.
+
+If this SYCL device is not a sub device or the sub device was not partitioned
+with ``sycl::info::partition_type::partition_by_affinity_domain``
+then the return value must be
+``sycl::info::partition_affinity_domain::not_applicable``,
+otherwise it must be one of the following values:
+``sycl::info::partition_affinity_domain::numa``,
+``sycl::info::partition_affinity_domain::L4_cache``,
+``sycl::info::partition_affinity_domain::L3_cache``,
+``sycl::info::partition_affinity_domain::L2_cache``,
+``sycl::info::partition_affinity_domain::L1_cache``.
+
++-------------------------------------------------------+
+| Return type: `sycl::info::partition_affinity_domain`_ |
++-------------------------------------------------------+
+
+
+Return types
+============
 
 .. _info-device_type:
 
 ``sycl::info::device_type``
-===========================
+---------------------------
 
 ::
 
+  namespace sycl::info {
+
   enum class device_type : unsigned int {
```

Comment:
```suggestion
  enum class device_type : /* unspecified */ {
```

The base type of these enumerations are all "unspecified" in the SYCL spec.  Same for the other enumerations below.